azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java

```diff
@@ -45,40 +52,57 @@
 
     @Override
     protected void doExecute() throws Exception {
-        createOrUpdateWebApp();
+        final ConfigurationParser parser = getParserBySchemaVersion();
+        final WebAppConfiguration webAppConfig = parser.getWebAppConfiguration();
+        createOrUpdateWebApp(webAppConfig);
         deployArtifacts();
     }
 
-    protected void createOrUpdateWebApp() throws Exception {
+    protected ConfigurationParser getParserBySchemaVersion() throws MojoExecutionException {
+        final String schemaVersion = StringUtils.isEmpty(getSchemaVersion()) ? "v1" : getSchemaVersion();
+
+        switch (schemaVersion.toLowerCase(Locale.ENGLISH)) {
+            case "v1":
+                return new V1ConfigurationParser(this);
+            case "v2":
+                return new V2ConfigurationParser(this);
+            default:
+                throw new MojoExecutionException(SchemaVersion.UNKNOWN_SCHEMA_VERSION);
+        }
+    }
+
+    protected void createOrUpdateWebApp(final WebAppConfiguration config) throws Exception {
         final WebApp app = getWebApp();

```

Comment:
To decouple the mojo and handler. we should not calling method from mojo

Reply:
Will add a todo item here and complete it in following PR(s).